File: Day 8/kotlin/main.kt
fun main() {
    val theString = "hello Racecar babb as nenonen"
    val result = findLongestPalindromes(theString)
    print(result)
}

fun isPalindrome(stringValue: String): Boolean {
    return (stringValue == stringValue.reversed())
}

fun findLongestPalindromes(theString: String): MutableSet<String> {
    val stringValue = theString.lowercase().replace(" ", "")

    var palindromesSet = mutableSetOf<String>()
    var longestLength = 0

    for (i in 0 until stringValue.length) {
        for (j in (i + 1)..stringValue.length) {
            val substring = stringValue.substring(i, j)
            if (isPalindrome(substring)) {
                val substringLength = substring.length
                if (substringLength > longestLength) {
                    longestLength = substringLength
                    palindromesSet = mutableSetOf(substring)
                }
                else if (substringLength == longestLength) {
                    palindromesSet.add(substring)
                }
            }
        }
    }

    return palindromesSet
}

File: Day 8/python/main.py
def is_palindrome(s):
    return s == s[::-1]

def longest_palindromes(the_string):
    the_string = the_string.lower().replace(" ", "")
    longest_pal_length = 0
    longest_palindromes_set = set()
    
    for i in range(len(the_string)):
        for j in range(i + 1, len(the_string) + 1):
            substring = the_string[i:j]
            if is_palindrome(substring):
                substring_length = len(substring)
                if substring_length > longest_pal_length:
                    longest_pal_length = substring_length
                    longest_palindromes_set = {substring}
                elif substring_length == longest_pal_length:
                    longest_palindromes_set.add(substring)
    
    return sorted(longest_palindromes_set)

the_string = "hello Racecar babb as nenonen"
result = longest_palindromes(the_string)
print("Longest palindromes:", result if result else "No palindromes found.")